File: examples/fleet.py
"""Write and recall a fact against a configured Caura backend."""

import uuid

from caura_rail import MemoryScope, Rail, RestMemoryStore, Visibility

marker = "Rail example " + str(uuid.uuid4())
scope = MemoryScope(agent_id="rail-example-writer", fleet_id="rail-examples", visibility=Visibility.TEAM)

with RestMemoryStore.from_env() as store:
    writer = Rail(store, scope)
    with writer.turn("Remember: " + marker) as turn:
        turn.reply = "Noted"
    if turn.degraded or not turn.writes or turn.writes[0].status != "written":
        raise RuntimeError("Write did not complete; inspect the turn errors and backend configuration")
    reader = Rail(store, scope.for_agent("rail-example-reader"))
    ctx = reader.recall(marker)
    if ctx.degraded or not any(marker in fact.content for fact in ctx.facts):
        raise RuntimeError("Second agent could not recall the fact; check fleet permissions")
    print("Second agent recalled the saved fact:", marker)

File: examples/fleet.mjs
import { randomUUID } from "node:crypto";
import { MemoryScope, Rail, RestMemoryStore } from "@caura/rail";

const marker = "Rail example " + randomUUID();
const store = RestMemoryStore.fromEnv(process.env);
const scope = new MemoryScope({
  agentId: "rail-example-writer", fleetId: "rail-examples", visibility: "scope_team",
});
const writer = new Rail({ store, scope });
const turn = await writer.turn("Remember: " + marker, () => "Noted");
if (turn.degraded || turn.writes[0]?.status !== "written") {
  throw new Error("Write did not complete; inspect the turn errors and backend configuration");
}
const reader = new Rail({ store, scope: scope.forAgent("rail-example-reader") });
const ctx = await reader.recall(marker);
if (ctx.degraded || !ctx.facts.some(f => f.content.includes(marker))) {
  throw new Error("Second agent could not recall the fact; check fleet permissions");
}
console.log("Second agent recalled the saved fact:", marker);
